dev/dll/XamlControlsResources.cpp

```diff
@@ -30,65 +47,51 @@ ThemeResources::ThemeResources()
-                else if (isRS4OrHigher)
-                {
-                    return L"ms-appx://" MUXCONTROLS_PACKAGE_NAME "/" MUXCONTROLSROOT_NAMESPACE_STR "/Themes/rs4_themeresources.xaml";
-                }
-                else if (isRS3OrHigher)
-                {
-                    return L"ms-appx://" MUXCONTROLS_PACKAGE_NAME "/" MUXCONTROLSROOT_NAMESPACE_STR "/Themes/rs3_themeresources.xaml";
-                }
-                else if (isRS2OrHigher)
-                {
-                    return L"ms-appx://" MUXCONTROLS_PACKAGE_NAME "/" MUXCONTROLSROOT_NAMESPACE_STR "/Themes/rs2_themeresources.xaml";
-                }
-                else
-                {
-                    return L"ms-appx://" MUXCONTROLS_PACKAGE_NAME "/" MUXCONTROLSROOT_NAMESPACE_STR "/Themes/rs1_themeresources.xaml";
-                }
+                packagePrefix = L"ms-appx://" MUXCONTROLS_PACKAGE_NAME "/" MUXCONTROLSROOT_NAMESPACE_STR  "/Themes/";
+            }
+
+            hstring releasePrefix;
+
+            if (is19H1OrHigher)
+            {
+                releasePrefix = L"19h1_";
+            }
+            else if (isRS5OrHigher)
+            {
+                releasePrefix = L"rs5_";
+            }
+            else if (isRS4OrHigher)
+            {
+                releasePrefix = L"rs4_";
+            }
+            else if (isRS3OrHigher)
+            {
+                releasePrefix = L"rs3_";
+            }
+            else if (isRS2OrHigher)
+            {
+                releasePrefix = L"rs2_";
             }
             else
             {
-                if (is19H1OrHigher)
-                {
-                    return L"ms-appx:///" MUXCONTROLSROOT_NAMESPACE_STR "/Themes/19h1_themeresources.xaml";
-                }
-                if (isRS5OrHigher)
-                {
-                    return L"ms-appx:///" MUXCONTROLSROOT_NAMESPACE_STR "/Themes/rs5_themeresources.xaml";
-                }
-                else if (isRS4OrHigher)
-                {
-                    return L"ms-appx:///" MUXCONTROLSROOT_NAMESPACE_STR "/Themes/rs4_themeresources.xaml";
-                }
-                else if (isRS3OrHigher)
-                {
-                    return L"ms-appx:///" MUXCONTROLSROOT_NAMESPACE_STR "/Themes/rs3_themeresources.xaml";
-                }
-                else if (isRS2OrHigher)
-                {
-                    return L"ms-appx:///" MUXCONTROLSROOT_NAMESPACE_STR "/Themes/rs2_themeresources.xaml";
-                }
-                else
-                {
-                    return L"ms-appx:///" MUXCONTROLSROOT_NAMESPACE_STR "/Themes/rs1_themeresources.xaml";
-                }
+                releasePrefix = L"rs1_";
             }
+
+            return packagePrefix + releasePrefix + compactPrefix + postfix;
         }()
     };
 
+    //  Prior to RS5, when ResourceDictionary.Source property is changed, we forgot to clear ThemeDictionaries.
+    ThemeDictionaries().Clear();
```

Comment:
What's this about? I didn't see it in the deleted files (although I could have missed it)

Reply:
Just like Comment say: Prior to RS5, when ResourceDictionary.Source property is changed, we forgot to clear ThemeDictionaries.
Old version doesn't need this because Source is only set once. but now we set Source twice and the default one first, the clear it and set Compact. We don't have function like OnApplyTemplate to indicate the dictionary is ready to use, so I didn't see a workaround

Reply:
The comment wasn't quite enough for me to understand. The extra detail you've provided here is helpful. Can you expand the comment in the code to include more of what you've said here?